streaming/src/main/scala/org/apache/spark/streaming/scheduler/BatchInfo.scala

```diff
@@ -36,8 +38,8 @@ case class BatchInfo(
     streamIdToInputInfo: Map[Int, StreamInputInfo],
     submissionTime: Long,
     processingStartTime: Option[Long],
-    processingEndTime: Option[Long]
-  ) {
+    processingEndTime: Option[Long],
+    private[streaming] val failureReasons: Map[Int, String] = Map.empty) {
```

Comment:
Wouldnt having another parameter in the constructor break binary compatibility, even if optional? I am scared of Scala doing crazy stuff.